sophiread/SophireadCLI/src/sophiread_core.cpp

```diff
@@ -0,0 +1,384 @@
+          const uint32_t width = tof_images[bin][0].size();
+          const uint32_t height = tof_images[bin].size();
+          std::vector<std::vector<TIFF32Bit>> accumulated_image = tof_images[bin];
+
+          // check if file already exist
+          if (std::filesystem::exists(filename)) {
+              TIFF* existing_tif = TIFFOpen(filename.c_str(), "r");
+              if (existing_tif) {
+                  uint32_t existing_width, existing_height;
+                  TIFFGetField(existing_tif, TIFFTAG_IMAGEWIDTH, &existing_width);
+                  TIFFGetField(existing_tif, TIFFTAG_IMAGELENGTH, &existing_height);
+
+                  if (existing_width == width && existing_height == height) {
+                      // Dimensions match, proceed with accumulation
+                      for (uint32_t row = 0; row < height; ++row) {
+                          std::vector<TIFF32Bit> scanline(width);
+                          TIFFReadScanline(existing_tif, scanline.data(), row);
+                          for (uint32_t col = 0; col < width; ++col) {
+                              accumulated_image[row][col] += scanline[col];
+                          }
+                      }
+                      spdlog::debug("Accumulated counts for existing file: {}", filename);
+                  } else {
+                      spdlog::error("Dimension mismatch for file: {}. Expected {}x{}, got {}x{}. Overwriting.",
+                                    filename, width, height, existing_width, existing_height);
+                  }
+                  TIFFClose(existing_tif);
+              } else {
+                  spdlog::error("Failed to open existing TIFF file for reading: {}", filename);
+              }
+          }
+
+          // Write or update TIFF file
+          TIFF* tif = TIFFOpen(filename.c_str(), "w");
+          if (tif) {
+              uint32_t width = tof_images[bin][0].size();
+              uint32_t height = tof_images[bin].size();
+
+              TIFFSetField(tif, TIFFTAG_IMAGEWIDTH, width);
+              TIFFSetField(tif, TIFFTAG_IMAGELENGTH, height);
+              TIFFSetField(tif, TIFFTAG_SAMPLESPERPIXEL, 1);
+              TIFFSetField(tif, TIFFTAG_BITSPERSAMPLE, 32);
+              TIFFSetField(tif, TIFFTAG_ORIENTATION, ORIENTATION_TOPLEFT);
+              TIFFSetField(tif, TIFFTAG_PLANARCONFIG, PLANARCONFIG_CONTIG);
+              TIFFSetField(tif, TIFFTAG_PHOTOMETRIC, PHOTOMETRIC_MINISBLACK);
+
+              for (uint32_t row = 0; row < height; ++row) {
+                  TIFFWriteScanline(tif, accumulated_image[row].data(), row);
+              }
+
+              TIFFClose(tif);
+              spdlog::debug("Wrote TIFF file: {}", filename);
+          } else {
+              spdlog::error("Failed to open TIFF file for writing: {}", filename);
+          }
+
+          // Accumulate counts for spectral file
+          spectral_counts[bin] = std::accumulate(accumulated_image.begin(), accumulated_image.end(), 0ULL,
```

Comment:
Another way to write this is
```c++
spectral_counts[bin] = std::accumulate(accumulated_image.begin(), accumulated_image.end(), static_cast<uint64_t>(0),
  [](const auto sum, const std::vector<unsigned int>& row) {
                return sum + std::accumulate(row.begin(), row.end(), 0ULL);
  });
```

The "interesting" bit is `const auto sum` in the lambda's argument list. The compiler will figure out the type based on the initial value, and declaring it `const` will never hurt.